source/WB-CAN-Firmware.c

```diff
@@ -166,28 +201,90 @@ int main(void) {
     for(;;);
 }
 
-/*******************************************************************************
- * Tasks
- ******************************************************************************/
-static void blinkTask(void *pv) {
-	while(1) {
 
-		vTaskDelay(pdMS_TO_TICKS(500));
-//		printf("BLINK");
+// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
+// Main Tasks:
+// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
+
+
+// Initialize Semaphores:
+SemaphoreHandle_t semaphore_Message;
+SemaphoreHandle_t semaphore_PWMActive;
+
+// Create Message Type that takes specific values using Enumerate:
+enum messageEnum
+    {
+        No_Command, DPR_Pause, DPR_Resume
+    };
+typedef enum messageEnum messageType;
```

Comment:
This can be shortened in the same way as the above enum. Also, I prefer to have each element of the enum on a new line. `messageType` will need to be changed in other bits of the code.

```suggestion
typedef enum {
        No_Command,
        DPR_Pause,
        DPR_Resume
    } message_t;
```

Reply:
Also, is this enum ever used outside the main task? If not, it can (should) be defined inside it.